Additional components for MCN v2.3 and MCN Breakout v1.0

aaronbeekay opened the pull request:
This PR adds the following components (as well as their associated entries in partsdb.txt):

### In Harwin.lbr
* **Device**: M22-25203xx (2x3 2mm pin hdr)
* **Device**: M22-71403xx (2x3 2mm female pin hdr)
* **Device**: M22-7131042 (1x10 2mm female pin hdr)
* **Device**: M22-2511005 (1x10 2mm pin hdr)

### In Diodes.lbr
* **Modify device**: add SOD-323 package to SCHOTTKY_DIODE device

### In Generic connectors.lbr
* **Device**: add screw-down binding posts

### In Molex.lbr
* **Device**: right-angle BNC jack

### In Generic footprints.lbr
* **Device**: right-angle DB-9 connector

aaronbeekay commented:
@sharrington93 - in order to exercise your new authority, can you review and merge this PR if you approve it?

sharrington93 commented:
@aaronbeekay I don't see any package or schematic for the binding posts change. Am I looking in the wrong place/misunderstand how this is used?

Otherwise, seems good.

aaronbeekay commented:
@sharrington93 Whoops. I was trying to delete some guidelines in d02c037, and it appears that I deleted the entire schematic and package instead and didn't notice it. I have reinserted the actual symbol and package, should be better now.

I also updated the symbols for the pin headers to better match the connector guidelines (they were generic symbols that didn't represent the mating face of the connector).

These commits triggers changes in Magic CAN Node.sch and MCN Breakout.sch, I'll add those separately.
